libjolie/src/main/java/jolie/lang/parse/module/Modules.java

```diff
@@ -36,13 +37,18 @@ public Map< URI, SymbolTable > symbolTables() {
 	}
 
 	public static ModuleParsedResult parseModule( ModuleParsingConfiguration configuration, InputStream stream,
-		URI programDirectory )
+		URI programURI )
 		throws ParserException, IOException, ModuleException {
 		ModuleParser parser = new ModuleParser( configuration );
-		ModuleFinder finder = new ModuleFinderImpl( configuration.packagePaths() );
-
 		ModuleRecord mainRecord = parser.parse(
-			new Scanner( stream, programDirectory, configuration.charset(), configuration.includeDocumentation() ) );
+			new Scanner( stream, programURI, configuration.charset(), configuration.includeDocumentation() ) );
+		ModuleFinder finder;
+
+		if( programURI.toString().contains( "jap:" ) ) {
+			finder = new ModuleFinderDummy();
```

Comment:
I don't really like this solution. I will think about the alternative later... The reason behind this now we need to pass the module path to finder. But Windows complain about having : on the path `jap:file`, so I create this Dummy to bypass the module lookup phase. As for now we do not support such thing in `jap` yet.

Reply:
I see! I'll add a todo to remember that we need to re-evaluate this one day.